libs/designer/src/lib/core/utils/parameters/segment.ts

```diff
@@ -106,12 +107,14 @@ export class ValueSegmentConvertor {
     return [this._createLiteralValueSegment(section)];
   }
 
-  private _convertStringToValueSegments(value: string): ValueSegment[] {
+  private _convertStringToValueSegments(value: string, parameterType: string): ValueSegment[] {
     if (isTemplateExpression(value)) {
       const expression = ExpressionParser.parseTemplateExpression(value);
       return this._convertTemplateExpressionToValueSegments(expression);
     }
-    return [this._createLiteralValueSegment(value)];
+    const isSpecialValue = ['true', 'false', 'null'].includes(value) || /^-?\d+$/.test(value);
+    const stringValue = parameterType === constants.SWAGGER.TYPE.ANY && isSpecialValue ? `"${value}"` : value;
```

Comment:
Please be careful of the cases when it is any field and user really wants to input non string

Reply:
This is for the function convertStringToValueSegment, where we first do a check of the value to be of type string. If the user has a different primitive it would not go into this logic.